Exiled.Events/Handlers/Internal/MapGenerated.cs

```diff
@@ -50,23 +50,20 @@ internal static class MapGenerated
         public static void OnMapGenerated()
         {
             Map.ClearCache();
-            Timing.CallDelayed(0.5f, () =>
-            {
-                GenerateCache();
-                Door.RegisterDoorTypesOnLevelLoad();
-            });
+            GenerateCache();
+            Door.RegisterDoorTypesOnLevelLoad();
         }
 
         private static void GenerateCache()
         {
+            GenerateTeslaGates();
```

Comment:
Wont this being before GenerateRooms cause TeslaGate.Room to be null?

Reply:
Yes it's that